superdesk/editor_utils.py

```diff
@@ -439,3 +644,20 @@ def prepend_block(self, block):
         else:
             index = 0
         self.blocks.insert(index, block)
+
+
+def replace_text(item, field, old, new, html=True):
+    editor = Editor3Content(item, field, html)
```

Comment:
It's a good idea to manage both editor2 and editor3, but I think we should have a dedicated `Editor2Content` which may simply inheritate from `Editor3Content` and remove `fields_meta` after updating.
There is already an `EditorContent` factory, you just have to replace the `NotImplementedError` by the new `Editor2Content`: https://github.com/superdesk/superdesk-core/blob/39dfe04397b99f95d88cb5e3437d1e4c3e734784/superdesk/editor_utils.py#L254

Reply:
I would actually keep fields_meta once it's parsed so we don't have to parse it again if doing multiple edits

Reply:
Editor instance is not returned, but still I find it more clean to use the dedicated factory, even if `Editor2Content` would only be a subclass of `Editor3Content`. This way, if one day we have `Editor4Content`, we would not have to change this method.